client/client-multi/client-objective-c/Kaa/DefaultEventManager.m

```diff
@@ -0,0 +1,262 @@
+- (void)clearState {
+    @synchronized(self.eventGuard) {
+        [self.currentEvents removeAllObjects];
+    }
+}
+
+- (void)produceEvent:(NSString *)eventFQN data:(NSData *)data target:(NSString *)target {
+    [self produceEvent:eventFQN data:data target:target transactionId:nil];
+}
+
+- (void)produceEvent:(NSString *)eventFQN data:(NSData *)data target:(NSString *)target transactionId:(TransactionId *)transactionId {
+    if (transactionId) {
+        DDLogInfo(@"%@ Adding event [eventClassFQN: %@, target: %@] to transaction %@", TAG, eventFQN, (target ? target : @"broadcast"), transactionId);
+        @synchronized(self.trxGuard) {
+            NSMutableArray *events = [self.transactions objectForKey:transactionId];
+            if (events) {
+                Event *event = [[Event alloc] init];
+                event.seqNum = -1;
+                event.eventClassFQN = eventFQN;
+                event.eventData = [NSData dataWithData:data];
+                if (target) {
+                    event.target = [KAAUnion unionWithBranch:KAA_UNION_STRING_OR_NULL_BRANCH_0 andData:target];
+                }
+                [events addObject:event];
+            } else {
+                DDLogWarn(@"%@ Transaction with id %@ is missing. Ignoring event.", TAG, transactionId);
+            }
+        }
+    } else {
+        DDLogInfo(@"%@ Producing event [eventClassFQN: %@, target: %@]", TAG, eventFQN, (target ? target : @"broadcast"));
+        @synchronized(self.eventGuard) {
+            Event *event = [[Event alloc] init];
+            event.seqNum = [self.state getAndIncrementEventSequenceNumber];
+            event.eventClassFQN = eventFQN;
+            event.eventData = [NSData dataWithData:data];
+            if (target) {
+                event.target = [KAAUnion unionWithBranch:KAA_UNION_STRING_OR_NULL_BRANCH_0 andData:target];
+            }
+            [self.currentEvents addObject:event];
+        }
+        
+        if (!self.isEngaged) {
+            [self.transport sync];
+        }
+    }
+}
+
+- (void)registerEventFamily:(id<BaseEventFamily>)eventFamily {
+    [self.registeredEventFamilies addObject:eventFamily];
+}
+
+- (void)onGenericEvent:(NSString *)eventFQN data:(NSData *)data source:(NSString *)source {
+    DDLogInfo(@"%@ Received event [eventClassFQN: %@]", TAG, eventFQN);
+    for (id<BaseEventFamily> family in self.registeredEventFamilies) {
+        DDLogInfo(@"%@ Lookup event fqn %@ in family %@", TAG, eventFQN, family);
+        if ([[family getSupportedEventFQNs] containsObject:eventFQN]) {
+            DDLogInfo(@"%@ Event fqn [%@] found in family [%@]", TAG, eventFQN, family);
+            [[self.executorContext getCallbackExecutor] addOperationWithBlock:^{
+                [family onGenericEvent:eventFQN withData:data from:source];
+            }];
+        }
+    }
+}
+
+- (NSInteger)findEventListeners:(NSArray *)eventFQNs delegate:(id<FindEventListenersDelegate>)delegate {
+    int requestId = arc4random();
```

Comment:
Can we you an incremental counter instead of a random generator as it is done in Endpoint Registration Manager?